Objects/bytearrayobject.c

```diff
@@ -1729,6 +1729,10 @@ bytearray_extend(PyByteArrayObject *self, PyObject *iterable_of_ints)
 
     while ((item = PyIter_Next(it)) != NULL) {
         if (! _getbytevalue(item, &value)) {
+            if (PyErr_ExceptionMatches(PyExc_TypeError) && PyUnicode_Check(iterable_of_ints)) {
+                PyErr_Format(PyExc_TypeError,
```

Comment:
It is not just `bytes`, any iterable of `int` is fine:

```python
>>> a = bytearray(b'')
>>> a.extend([1, 2, 3])
>>> a
bytearray(b'\x01\x02\x03')
```

Typeshed's definition: https://github.com/python/typeshed/blob/a3d356c04c15afc22b254afae2e6e7adbbe031a5/stdlib/builtins.pyi#L726

Reply:
`int`-like maybe? 🤔 
I propose searching through existing code for similar types to make it consistent.

Reply:
>`int`-like

I prefer this actually. However, I do not find any error messages in the repo containing something similar; I can only find documentation (and not many). I searched for int-like, 'int'-like, or integer-like through VS Code.

There are a lot of mentions for integer though. I guess I will just stick to integer then. Anyway, I think it matches the original error message. `TypeError: 'str' object cannot be interpreted as an integer`. It was not completely wrong; it was just not very clear.

Reply:
Before saying 'integers' perhaps a subclass Int of int should be tested.  In the example above, `a.extend([Int(1), Int(2)])`.  The ints are required to be in range 0-255, but we do not do value typing.  I believe an int out of that range gets a separate message.

Reply:
This is also true. But I'm not really sure what a good catch-all term is. If we say bytes or 8-bit integers, people may think `int` cannot be used.

Unless we spell it all out, and say something like `Expected iterable of integers (each from 0 to 255); got: 'str'`. However, it won't be very concise. I guess it's the most clear though, and maybe clarity is the most important.

Reply:
@terryjreedy Yea, it does already raise an error: `ValueError: byte must be in range(0, 256)`. This can be found in the function `_getbytevalue`. This is also already tested in the function `test_extend` of the class `test.test_bytes.ByteArrayTest` from `Lib/test/test_bytes.py`. Would this be sufficient?

Reply:
@sobolevn @terryjreedy Hi, may I ask if there are any other concerns you have about this? Feel free to let me know if there is anything you would still like me to change.

Reply:
My remaining concern is elsewhere.